ports/esp32/modesp32.c

```diff
@@ -177,7 +188,18 @@ static mp_obj_t esp32_wake_on_gpio(size_t n_args, const mp_obj_t *pos_args, mp_m
     if (args[ARG_pins].u_obj != mp_const_none) {
         size_t len = 0;
         mp_obj_t *elem;
-        mp_obj_get_array(args[ARG_pins].u_obj, &len, &elem);
+        mp_obj_t pins_obj = args[ARG_pins].u_obj;
```

Comment:
Instead of copying this block of code in two places, could you extract the common parts to a static helper function and call that?

i.e. `get_pins_arg(args[ARG_pins].u_obj, &len, &elem)` ?